src/NuGet.Core/NuGet.Versioning/FloatRange.cs

```diff
@@ -423,6 +424,51 @@ public override string ToString()
             return result;
         }
 
+        /// <summary>
+        /// Create a floating version string in the format: 1.0.0-alpha-*
+        /// </summary>
+        public void ToString(StringBuilder sb)
```

Comment:
Does this have to be public for LibraryModel to call it?

Reply:
Yes. Apparently IVTs aren't a thing in this repo.

Reply:
Can we avoid duplicating the tostring logic?

For example, I think we could have the `ToString()` method call this one with it's own stringbuilder.

Reply:
My hunch while writing this originally was that doing so would increase allocations in the `ToString()` method. But looking now inside mscorlib I see that `string.Format` acquires a pooled `StringBuilder` to perform the format. So we can do the same on our side and reduce the duplication. Update pushed.